index.js

```diff
@@ -117,8 +118,30 @@ const cleanUpLinks = (input) => {
   return input
 }
 
+const normalizeLink = (entityMap, link, removeUnresolved) => {
+  const resolvedLink = getResolvedLink(entityMap, link)
+  if (resolvedLink === UNRESOLVED_LINK) {
+    return removeUnresolved ? resolvedLink : link
+  }
+  return resolvedLink
+}
+
+const maybeNormalizeLink = (maybeLink, entityMap, removeUnresolved) => {
+  if (Array.isArray(maybeLink)) {
+    maybeLink = maybeLink.map((maybeArrayLink) => maybeNormalizeLink(maybeArrayLink, entityMap, removeUnresolved))
```

Comment:
Typically not a fan of mutating input params.   Is there a way do this with immutable params?

Reply:
agreed...I got caught up copying that style that was already present in other functions here. fixed here: https://github.com/contentful/contentful-resolve-response/pull/377/commits/cc3618e68642cf75c0a9c6e4250121e9519ebf09